src/servers/src/influxdb.rs

```diff
@@ -141,14 +139,252 @@ impl TryFrom<&InfluxdbRequest> for InsertRequests {
+            ),
+            FieldValue::Boolean(v) => (ColumnDataType::Boolean, ValueData::BoolValue(*v)),
+        };
+
+        if *index == schema.len() {
+            schema.push(ColumnSchema {
+                column_name: k.to_string(),
+                datatype: datatype as i32,
+                semantic_type: SemanticType::Field as i32,
+            });
+            one_row.push(to_value(value));
+        } else {
+            check_schema(datatype, SemanticType::Field, &schema[*index])?;
+            // unwrap is safe here
+            one_row.get_mut(*index).unwrap().value_data = Some(value);
+        }
+    }
+
+    Ok(())
+}
+
+fn parse_ts(
+    ts: Option<i64>,
+    precision: Option<Precision>,
+    column_indexes: &mut HashMap<&str, usize>,
+    schema: &mut Vec<ColumnSchema>,
+    one_row: &mut Vec<Value>,
+) -> Result<(), Error> {
+    let precision = unwrap_or_default_precision(precision);
+    let ts = match ts {
+        Some(timestamp) => writer::to_ms_ts(precision, timestamp),
+        None => {
+            let timestamp = Timestamp::current_millis();
+            let unit = get_time_unit(precision)?;
+            let timestamp = timestamp
+                .convert_to(unit)
+                .with_context(|| TimePrecisionSnafu {
+                    name: precision.to_string(),
+                })?;
+            writer::to_ms_ts(precision, timestamp.into())
+        }
+    };
+
+    let column_name = INFLUXDB_TIMESTAMP_COLUMN_NAME;
+    let index = column_indexes.entry(column_name).or_insert(schema.len());
+    if *index == schema.len() {
+        schema.push(ColumnSchema {
+            column_name: column_name.to_string(),
+            datatype: ColumnDataType::TimestampMillisecond as i32,
+            semantic_type: SemanticType::Timestamp as i32,
+        });
+        one_row.push(to_value(ValueData::TsMillisecondValue(ts)))
+    } else {
+        check_schema(
+            ColumnDataType::TimestampMillisecond,
+            SemanticType::Timestamp,
+            &schema[*index],
+        )?;
+        // unwrap is safe here
+        one_row.get_mut(*index).unwrap().value_data = Some(ValueData::TsMillisecondValue(ts));
+    }
+
+    Ok(())
+}
+
+#[inline]
+fn check_schema(
+    datatype: ColumnDataType,
+    semantic_type: SemanticType,
+    schema: &ColumnSchema,
+) -> Result<(), Error> {
+    ensure!(
+        schema.datatype == datatype as i32,
+        IncompatibleSchemaSnafu {
+            column_name: &schema.column_name,
+            datatype: "datatype",
+            expected: schema.datatype,
+            actual: datatype as i32,
+        }
+    );
+
+    ensure!(
+        schema.semantic_type == semantic_type as i32,
+        IncompatibleSchemaSnafu {
+            column_name: &schema.column_name,
+            datatype: "semantic_type",
+            expected: schema.semantic_type,
+            actual: semantic_type as i32,
+        }
+    );
+
+    Ok(())
+}
+
+#[inline]
+fn to_value(value: ValueData) -> Value {
```

Comment:
suggestion(non-blocking): Maybe we could implement `From<ValueData> for Value` in `greptime-proto`.

Reply:
Very good suggestion, I will complete it in the next PR since the code needs to be written in the greptime-proto module. https://github.com/GreptimeTeam/greptime-proto/pull/82